Seed Database w/ Dummy Team and Judge Data

OlinZ opened the pull request:
## Pre-Requisites
- [x] Yes, I updated [Authors.md](../blob/main/Authors.md) **OR** this is not my first contribution
- [ ] Yes, I included and/or modified tests to cover relevent code **OR** my change is non-technical
- [x] Yes, I wrote this code entirely myself **OR** I properly attributed these changes in [Third Party Notices](../blob/main/THIRD-PARTY-NOTICES.txt)

<!-- After addressing the pre-requisites above, make sure to fill out BOTH sections below -->
<!-- NOTE: This is a comment; the comments below will be hidden when you submit -->

## Description of Changes
<!-- Enter a description of what this PR adds/changes -->
Used [typeorm-seeding](https://github.com/w3tecch/typeorm-seeding) to seed db with fake team and judge data. We had to change `"module": "esnext"` to `"module": "commonjs"` so `typeorm-seeding` compiles and runs (not sure if this is the right solution). 

To seed the db: run `npm run seed:run` and it will generate 10 fake teams and 10 fake judges.

## Related Issues
<!-- Include a list and brief description of any tracked issues -->
<!-- NOTE: Make sure to use the `Closes`/`Fixes` syntax to automatically close the issue when your PR is merged -->
<!-- More detail: https://docs.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords -->
<!-- e.g., "Closes #123 - A bug that crashes the app" -->
Related to #260 

`Co-authored-by: Evelio Ramos <44717805+eveliososa@users.noreply.github.com>`

johnkahn commented:
Looks good, what was the error that you were getting before you changed the module?

OlinZ commented:
> Looks good, what was the error that you were getting before you changed the module?

Something about `Cannot use import statement outside a module`. Here's a screenshot of the error.

![image](https://user-images.githubusercontent.com/42856711/113789075-13a25080-9704-11eb-9745-8d7b486a6b45.png)

johnkahn commented:
Gotcha, so it's because those files don't export anything. Could you put the `define` code in the entity itself? Like at the bottom of the file, I think you'll need to modify the configuration in the ormconfig file to point the seed factories to the entities instead of the factories. But I think that'll work and let us keep `esnext` as our module type.

OlinZ commented:
I tried that and it still throws the same error.
Here's what I did:
- Moved the contents of `judge.factory.ts` and `team.factory.ts` to below the code in `src/entities/team.ts` and `src/entities/judge.ts`.
- Added `"factories": ["src/entities/team.ts", "src/entities/judge.ts"]` to `ormconfig.json`.
- Changed `"module": "commonjs"` back to "module": "esnext"

Here's the error:
![image](https://user-images.githubusercontent.com/42856711/113920334-4141d500-97aa-11eb-8dd5-ec49e3922a6d.png)

johnkahn commented on src/seed.ts:
```diff
@@ -0,0 +1,82 @@
+import { env } from './env';
+import logger from './logger';
+
+async function initDatabase(): Promise<void> {
```
You should be able to just import and call the same initDatabase that the app does so you don't need to write your own

OlinZ replied:
I think don't the code exports `initDatabase` anywhere. I just copied this function from `app.ts`. I guess we can export `initDatabse` in `app.ts` and import it here, let me know what you think.

johnkahn replied:
Gotcha, yeah let's export that and use it here 👍

johnkahn reviewed (changes requested):
I think it's good to go, just a small change, and it's hard to review because of all the code cov warnings, can you add `!src/seed.ts` to the coverage ignore section of `jest.config.js` file? That will make it easier to review.

![image](https://user-images.githubusercontent.com/7807353/115065945-72aa5700-9eb4-11eb-9170-5009481eb0b7.png)
